Update LNM cluster presets

davidrudlstorfer opened the pull request:
<!--
* Title: Provide a general summary of your changes in the Title above.

* Assignees:  If you know anyone who should likely handle bringing this pull request (PR) to completion -- that's usually yourself -- select them from the Assignees drop-down on the right.

* Labels: Update the label of the issue(s) addressed by this pull request to "Under Review".
-->

## Description and Context
<!--
Provide a brief and concise description of your proposed change. Questions you should think about:
* Why is this change required?  What problem does it solve?
* Is there a bigger picture? Is this PR a part of a larger set of changes? Which further steps are planned after merging this PR, if any?
* How has the proposed implementation been verified and tested?

Keep the description of the PR always up-to-date and concise.
-->

Update the LNM Cluster config to actually use the new GCC 13.2 compiler

Also add FOUR_C_OPENMPI_ROOT to remove the necessity that each user loads the module within their bash.rc file

## Related Issues and Merge Requests
<!--
If applicable, let us know how this pull request is related to any other open issues or pull requests by linking to them here.
Some suggestion for keywords:
Closes (will automatically close mentioned issue if merged), Blocks, Related to
-->

davidrudlstorfer commented:
@sebproell regarding our short discussion - the user still has to load cmake 3.30.5 in their bash.rc file

sebproell commented on presets/lnm/cluster/CMakePresets.json:
```diff
@@ -23,8 +23,9 @@
       "cacheVariables": {
-        "CMAKE_CXX_COMPILER": "/cluster/mpi/gcc/openmpi/4.1.5/bin/mpic++",
-        "CMAKE_C_COMPILER": "/cluster/mpi/gcc/openmpi/4.1.5/bin/mpicc",
+        "FOUR_C_OPENMPI_ROOT": "/opt/openmpi/4.1.5/",
```
```suggestion
        "FOUR_C_MPI_ROOT": "/opt/openmpi/4.1.5/",
```

davidrudlstorfer replied:
Thanks - changed it

sebproell commented on presets/lnm/cluster/CMakePresets.json:
```diff
@@ -23,8 +23,9 @@
       "cacheVariables": {
-        "CMAKE_CXX_COMPILER": "/cluster/mpi/gcc/openmpi/4.1.5/bin/mpic++",
-        "CMAKE_C_COMPILER": "/cluster/mpi/gcc/openmpi/4.1.5/bin/mpicc",
+        "FOUR_C_MPI_ROOT": "/opt/openmpi/4.1.5/",
```
Did you accidentally flip the settings for thought and bruteforce?

davidrudlstorfer replied:
Damnit :joy: thanks for further looking into it - changed it - hopefully now its correct

davidrudlstorfer commented:
Unfortunately, both cluster configs failed and I am rather clueless why it fails. Maybe someone (@sebproell, @amgebauer) has a clue?

https://gitlab.lrz.de/baci/baci-config-lnm/-/pipelines/2298444

sebproell commented:
@davidrudlstorfer the reason is that the system-installed `libstc++` is used when executing 4C. @amgebauer and I kind of suspected that but thought we should give it a try. This is a frequent problem when building another compiler and trying to compile executables with it that remember which `libstc++` should be used. We can also go with the `module load` solution instead. In this case, we would drop all the paths and require a proper `module load` in CI and by our coworkers.

davidrudlstorfer commented:
> @davidrudlstorfer the reason is that the system-installed libstc++ is used when executing 4C. @amgebauer and I kind of suspected that but thought we should give it a try. This is a frequent problem when building another compiler and trying to compile executables with it that remember which libstc++ should be used. We can also go with the module load solution instead. In this case, we would drop all the paths and require a proper module load in CI and by our coworkers.

Thanks for the explanation - one question remains - you write that the problem happens during execution but it actually fails during linking - shouldn't this work?

For me the module load approach is fine - so let's just stick with it if it works and remove all the unnecessary variables from the presets.

sebproell commented:
>  Thanks for the explanation - one question remains - you write that the problem happens during execution but it actually fails during linking - shouldn't this work?

I did not check carefully and thought the problem appears once we execute 4C to generate the metadata file. You are right it happens during linking which is ...  interesting.

davidrudlstorfer commented:
As discussed I've now reverted the approach to the old one, i.e., load all modules within the bashrc file and I've removed the hardcoded paths to the compilers from the cluster preset

I started a pipeline here https://gitlab.lrz.de/baci/baci-config-lnm/-/pipelines/2299473

Let's see if it works now...

davidrudlstorfer commented:
@amgebauer @sebproell  the pipeline succeeded on `bruteforce` and `thought`- therefore, this can be merged to finally fix the LNM pipeline

Once this is merged I will inform all colleagues regarding the module update within their .bashrc file

amgebauer reviewed (approved):
Good that we will have these changes in our first release.
